### ISSUE TITLE
Build Error - Makefile.am

### ISSUE THREAD

**Issue by gene-git** [Motion-Project/motion#1013]
Linux Arch - fully updated

I build from the master branch of the git repo.
I started getting build errors for the install phase.
i.e make install 
I "fixed" them locally by removing a few lines from Makefile.am before running configure.

This patch works for me, though there well may be a better approach.

[make.patch.txt](https://github.com/Motion-Project/motion/files/3643289/make.patch.txt)

**Comment by Mr-DaveDev** [comment]
We would need to see the actual error messages since we do not have a readily available Arch distro platform.

The `make install` was however tested on our usual distros (apple, alpine, Ubuntu, Debian, FreeBSD and Raspian) without issue.  

The particular section removed is the install and update of the sample configuration files.

**Comment by Mr-DaveDev** [comment]
It must be the `DESTDIR` .  

I would note that the lines are not duplicated and are different.  The first set applies the conversion to the configuration files that were installed in the `sysconfdir`.  The second applies them to the files in the data directory.  

If you review the sample files in your `DESTDIR` I would guess you will find that they still include `prefix`

**Comment by gene-git** [comment]
You're right - I missed that - pretty clear from Makefile too.

Think you can get DESTDIR working again - obviously important for building packages to have that functionality.

thanks!

**Comment by Mr-DaveDev** [comment]
Look at the very bottom of the `motion-dist.conf` and you will see
`; camera ${prefix}/etc/motion/camera1.conf` 
Likewise at the top of each `camera-dist.conf` there will be a 
`# ${prefix}/etc/motion/camera1.conf`

The revision to prefix with DESTDIR has been pushed in.

**Comment by gene-git** [comment]
Hah - right you are - i do see things like:

; camera_dir @sysconfdir@/motion/conf.d

However now I am getting a different error:
/bin/sh: /home/build/motion/pkg/motion/etc/motion/motion-dist.conf.tmp: No such file or directory

This dir is of course completely empty when the build starts.
It seems like it is trying to do something with files from that install dir before they are created.

**Comment by gene-git** [comment]
Running make -d helps me understand what is actually happening. 

I may well be missing something - but from what I see, the file data/motion-dist.conf is created by system build from data/motion-dist.conf.in  by doing:

- ./config.status data/motion-dist.conf

This has all the substitutions for sysconfdir etc already done - ( file has no @sysconfdir@ and no $prefix refs. )

So unclear to me what the sed command is doing - but nonetheless,  should the "source" file for the sed command not be "data/motion-dist.conf" rather than "${DESTDIR}${sysconfdir}/motion-dist.conf" .

The latter of course does not exist (unless build is run where the destination is already populated, which is not the case when building packages with clean/empty destination).

So, should the sed command not be changed to use this source file - i.e.

- @sed -e 's|$${prefix}|$(prefix)|' data/motion-dist.conf > ${DESTDIR}${sysconfdir}/motion-dist.conf.tmp ....

Same for the others.

**Comment by gene-git** [comment]
I've found another issue with the build setup - to make things work I patched Makefile.am as in previous comment.

Now make DESTDIR=x install succeeds
however make -j8 DESTDIR=x install can fail

It is  a race regarding the sample files - since each line is separate shell command - make tries to run them in parallel which can fail - probably anything related to any single sample config file should be a single shell line in the Makefile instead of multiple lines to avoid the race when doing parallel make.

both builds started from scratch in empty build dir and begin with fresh clone of repo and build.

**Comment by gene-git** [comment]
This patch seems to work for me, perhaps its flawed or perhaps there is a better way.
For now using this patch lets me build motion from git master again.

[make.patch.txt](https://github.com/Motion-Project/motion/files/3654384/make.patch.txt)

Though it still looks like some of the sed commands are no ops as the autoconf system has already done the substitution - but I could be wrong.

**Comment by gene-git** [comment]
Running configure definitely updates the files in data and pretty sure the entire sed line can be removed and all that needs to be done is copy the sample files to the right destination.

[make.patch.txt](https://github.com/Motion-Project/motion/files/3654836/make.patch.txt)

**Comment by Mr-DaveDev** [comment]
I would suggest not looking at the files in the data directory but instead those in the install location.  Once you comment out the first section of `sed` commands you will be able to see the issue.  Or possibly just remove the whole `install-data-local` rule.  

What I observed is the default install without any local rules will copy the files but not do any updates to them to change the `${prefix}`.  So the `install-data-local` rule was added but unfortunately, it runs AFTER the files have been copied from the data directory.

**Comment by gene-git** [comment]
Sigh - sorry trying to help ... good thing its only the samples. Hopefully you have a fix that will actually do the right thing :)

**Comment by gene-git** [comment]
With the patch I'm using which simply copies the files from data to the destination dir - what I observe is that :
(1) running configure - updates the files in data dir and they all look correct for the configured sysconf and prefix
(2) the subsequent make install runs the install-data-local rule (which in my version)  just copies files from data to the install sysconfdir. They are correct in both places (data and install) - by construction the files are identical.

So presumably I'm missing what is actually wrong as it "seems" to be doing the correct thing.
At least for me.

**Comment by gene-git** [comment]
Current git master works fine for me ... looks like you removed the troublesome section.

thanks!
